### ISSUE TITLE
Cannot use import statement outside a module

### ISSUE THREAD

**Issue by LOLBRUHNICE** [RIAEvangelist/node-ipc#217]
When i try to run my application that use node-ipc, i get an error:
Cannot use import statement outside a module
```
D:\xxx\.yarn\cache\node-ipc-npm-10.0.2-1402b3557b-8f221780e6.zip\node_modules\node-ipc\node-ipc.js:1
import IPC from './services/IPC.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
 ...
```
it happen both in js or ts

**Comment by RIAEvangelist** [comment]
You need to put "type":"module" in your package.json

**Comment by LOLBRUHNICE** [comment]
i get this error after i change type from commonjs to module:
```
Object.defineProperty(exports, "__esModule", { value: true });
                      ^

ReferenceError: exports is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and 'D:\xxx\package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
```
(compiled with ESNEXT)

**Comment by RIAEvangelist** [comment]
The way you do exports is different in es modules. You use export instead.

You could also exclude it from your bundle and import it as a regular
module instead.

**Comment by LOLBRUHNICE** [comment]
i compile my file with tsc and tsc always use 
```js
Object.defineProperty(exports, "__esModule", { value: true });
```
how can i use export instead of exports?

**Comment by RIAEvangelist** [comment]
fixed in PR #219 by @tripodsgames will push the latest version 10.2.0 shortly.

**Comment by joshuaalayon** [comment]
Be aware - this package was found to contain malware that detects its users machines purported geographical location and replaces the data from the machine if it does not like the location it detects

If you are still using this package - you may do better by locking it to an older version or, preferably, replacing it altogether along with all software maintained by the repos maintainer(s)

https://www.bleepingcomputer.com/news/security/big-sabotage-famous-npm-package-deletes-files-to-protest-ukraine-war/amp/
